### PR TITLE
Add coupon system

### PR DESCRIPTION
**Pull request by TheFallenSpirit**

### CODE REVIEW

**Comment by heyitslumi** [comment]
epic

File: app/Console/Commands/Schedule/CouponCommand.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        $coupons = Coupon::query()->get();
+        foreach ($coupons as $coupon) {
+            $carbon = new Carbon($coupon->expires);
+            $expires = $carbon->timestamp;
+            if ($time >= $expires) {
+                $coupon->update(['expired' => true]);
+                $this->line('Coupon #'.$coupon->id.' has been set as expired.');
+            }
+        }
+        $this->line('Completed check for expired coupons.');
```
Comment (camwhi-te) [inline app/Console/Commands/Schedule/CouponCommand.php]: PHP-CS-Fixer will not like this spacing - also, there is no use for all of the local vars

File: app/Http/Controllers/Api/Client/AccountController.php [hunk 2]
```diff
@@ -106,6 +108,28 @@ public function verify(Request $request): JsonResponse
...
+        if ($coupon->getAttribute('expired')) {
+            throw new DisplayException('This coupon has expired.');
+        }
+        if ($coupon->getAttribute('uses') < 1) {
+            throw new DisplayException('This coupon has no uses left.');
+        }
+        $balance = $request->user()->store_balance;
+        $request->user()->update(['store_balance' => $balance + $coupon->cr_amount]);
+        Coupon::query()->where('code', $code)->update(['uses' => $coupon->uses - 1]);
+        return new JsonResponse([], Response::HTTP_NO_CONTENT);
```
Comment (camwhi-te) [inline app/Http/Controllers/Api/Client/AccountController.php]: PHP-CS-Fixer will fix this

**Review by camwhi-te — CHANGES_REQUESTED** [review]
Please reference comments as well as run Eslint + PHP-CS-Fixer to stop the workflows failing, 👍🏼

File: app/Console/Commands/Schedule/CouponCommand.php (hunk 1 above)
Comment (camwhi-te) [reply app/Console/Commands/Schedule/CouponCommand.php]: if ($time >= new Carbon($coupon->expires)->timestamp) { ... }

Comment (TheFallenSpirit) [reply app/Console/Commands/Schedule/CouponCommand.php]: > 

image
Attempted this and it failed.

**Comment by TheFallenSpirit** [comment]
I ran PHPCS Fixer (vendor/bin/php-cs-fixer fix), it said it fixed all the issues locally. Also the code that is now failing is not my code.

**Pull request merged**